server/src/main/java/org/elasticsearch/action/support/TransportAction.java

```diff
@@ -40,12 +40,19 @@ protected TransportAction(String actionName, ActionFilters actionFilters, TaskMa
      * Use this method when the transport action should continue to run in the context of the current task
      */
     public final void execute(Task task, Request request, ActionListener<Response> listener) {
-        ActionRequestValidationException validationException = request.validate();
+        final ActionRequestValidationException validationException;
+        try {
+            validationException = request.validate();
+        } catch (Exception e) {
```

Comment:
I am trying to understand this part of code. If assertion always `false`, it means we will always throw `AssertionError`, right?  So the code after `assert false` will never be executed?

Or are we checking assertions only during testing mode?

Reply:
@mayya-sharipova Here we fail with AssertionError in tests, but log a warning message and fail the request with an exception in production.

Reply:
@dnhatn Thanks for clarifying, I was also thinking the same.